storm-core/src/clj/backtype/storm/daemon/nimbus.clj

```diff
@@ -984,25 +1094,30 @@
         ))))
 
 (defn cleanup-corrupt-topologies! [nimbus]
-  (if (is-leader nimbus :throw-exception false)
-    (let [storm-cluster-state (:storm-cluster-state nimbus)
-          code-ids (set (code-ids (:conf nimbus)))
-          active-topologies (set (.active-storms storm-cluster-state))
-          corrupt-topologies (set/difference active-topologies code-ids)]
-      (doseq [corrupt corrupt-topologies]
-        (log-message "Corrupt topology " corrupt " has state on zookeeper but doesn't have a local dir on Nimbus. Cleaning up...")
-        (.remove-storm! storm-cluster-state corrupt)
-        )))
-  (log-message "not a leader, skipping cleanup-corrupt-topologies"))
-
-;;setsup code distributor entries for all current topologies for which code is available locally.
-(defn setup-code-distributor [nimbus]
   (let [storm-cluster-state (:storm-cluster-state nimbus)
-        locally-available-storm-ids (set (code-ids (:conf nimbus)))
+        blob-store (:blob-store nimbus)
+        code-ids (set (code-ids blob-store))
         active-topologies (set (.active-storms storm-cluster-state))
-        locally-available-active-storm-ids (set/intersection locally-available-storm-ids active-topologies)]
-    (doseq [storm-id locally-available-active-storm-ids]
-      (.setup-code-distributor! storm-cluster-state storm-id (:nimbus-host-port-info nimbus)))))
+        corrupt-topologies (set/difference active-topologies code-ids)]
+    (doseq [corrupt corrupt-topologies]
+      (log-message "Corrupt topology " corrupt " has state on zookeeper but doesn't have a local dir on Nimbus. Cleaning up...")
+      (.remove-storm! storm-cluster-state corrupt)
+      (if (instance? LocalFsBlobStore blob-store)
+        (doseq [blob-key (get-key-list-from-id (:conf nimbus) corrupt)]
+          (.remove-blobstore-key! storm-cluster-state blob-key)))
+      )))
+
+;;setsup blobstore for all current keys
+(defn setup-blobstore [nimbus]
+  (let [storm-cluster-state (:storm-cluster-state nimbus)
+        blob-store (:blob-store nimbus)
+        local-list-of-keys (set (get-key-list-from-blob-store blob-store))
```

Comment:
In `get-key-list-from-blob-store`, you convert to a Seq, then to a vector, here to a set, and below to an ArrayList.